File: problem-solving/baekjoon/11653_소인수분해.py
# https://www.acmicpc.net/problem/11653
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    factor = 2
    while factor <= n:
        if n%factor == 0:
            n /= factor
            print(factor)
        else:
            factor += 1

File: problem-solving/baekjoon/11653_소인수분해.go
// https://www.acmicpc.net/problem/11653
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	for factor := 2; factor <= n; {
		if n%factor == 0 {
			n /= factor
			fmt.Fprintln(writer, factor)
		} else {
			factor++
		}
	}
}
